frontend/pages/policies/ManagePoliciesPage/components/InstallSoftwareModal/InstallSoftwareModal.tsx

```diff
@@ -152,19 +158,62 @@ const InstallSoftwareModal = ({
     [formData]
   );
 
-  const availableSoftwareOptions = titlesAFI?.map((title) => {
-    const splitName = title.software_package?.name.split(".") ?? "";
-    const ext =
-      splitName.length > 1 ? splitName[splitName.length - 1] : undefined;
-    const platformString = ext
-      ? `${getPlatformDisplayFromPackageExtension(ext)} (.${ext}) • `
-      : "";
-    return {
-      label: title.name,
-      value: title.id,
-      helpText: `${platformString}${title.software_package?.version ?? ""}`,
-    };
-  });
+  /**
+   * Filters and transforms software titles into dropdown options
+   * Filters titlesAFI to include only software compatible with the policy's
+   * platform(s) by using the file extension to determine the platform
+   */
+  const availableSoftwareOptions = (policy: IFormPolicy) => {
+    return titlesAFI
+      ?.filter((title) => {
+        const splitName = title.software_package?.name.split(".") ?? "";
+        const ext =
+          splitName.length > 1 ? splitName[splitName.length - 1] : undefined;
+        const platform = getPlatformFromExtension(ext);
+
+        // Filter apps to FMA/custom packages with extensions that match policy's platform
+        const isAvailableSoftwarePackageOption =
+          platform && policy.platform.split(",").includes(platform);
+
+        // Filters apps to only macOS VPP apps for macOS policies
+        const isAvailableVPPOption =
+          policy.platform.split(",").includes("darwin") &&
+          title.source === "apps" &&
+          !!title.app_store_app;
+
+        return isAvailableSoftwarePackageOption || isAvailableVPPOption;
+      })
+      .map((title) => {
+        const vppOption = title.source === "apps" && !!title.app_store_app;
```

Comment:
Seems like this would wind up simpler if we handled all of the VPP logic behind an if + early return and then fell back to the installer logic if we got there. Means duplicating L210-212 but I think that in return we'll be able to get rid of a few lines of ternaries and get the code a bi cleaner.